### ISSUE TITLE
Error in Test.hs: Non-exhaustive patterns in record update 

### ISSUE THREAD

**Issue by keremispirli** [TorXakis/TorXakis#44]
Started TorXakis for the example *adder* with PAdder.txs model definition.
Ran following commands:
```
tester Adder Purp1 Sut
test 4
```
Got error:
Test.hs:(421,38)-(426,23): Non-exhaustive patterns in record update

Failing code:
(421,38)-(426,23):
```
then modify $ \env -> env
       { IOC.behtrie  = (IOC.behtrie env) ++
                        [ ( IOC.curstate env, act, (IOC.maxstate env)+1 ) ]
       , IOC.curstate = (IOC.maxstate env)+1
       , IOC.maxstate = (IOC.maxstate env)+1
       }
```
Error log: [txs.9876.err.log.txt](https://github.com/TorXakis/TorXakis/files/1164897/txs.9876.err.log.txt)

**Comment by keremispirli** [comment]
Same error at (443,42)-(448,27):
```
                   then modify $ \env -> env
                          { IOC.behtrie  = (IOC.behtrie env) ++
                                           [ ( IOC.curstate env, act, (IOC.maxstate env)+1 ) ]
                          , IOC.curstate = (IOC.maxstate env)+1
                          , IOC.maxstate = (IOC.maxstate env)+1
                          }
```

Duplicate code?

**Comment by pjljvandelaar** [comment]
In Testing mode, maxstate is not defined!

maxstate is only defined in Stepping, yet txsSetStep (of TxsCore) only accepts a Model (not a TestPurpose!)

**Comment by ramon-janssen** [comment]
As for testing, maxstate seems to be always equal to curstate, so updates of maxstate indeed seem redundant. Removing this update and replacing any occurrences of maxstate by curstate in Test.hs fixes it, except I then still have issue #58 (I don't know if it is related, and perhaps even caused by this fix).

**Comment by keremispirli** [comment]
Looks like this is fixed now.
